### PR TITLE
Adding typings for requestretry

### PR DESCRIPTION
**Pull request by EricByers**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/requestretry/tslint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "dtslint/dt.json",
+  "rules": {
+    "ban-types": false
```
Comment (andy-ms) [inline types/requestretry/tslint.json]: What errors was this giving you?

File: types/requestretry/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+	interface RequestRetryOptions extends request.CoreOptions {
+		maxAttempts?: number;
+		retryDelay?: number;
+		retryStrategy?: RetryStrategy;
+	}
+
+	type OptionsWithUri = request.UriOptions & RequestRetryOptions;
+	type OptionsWithUrl = request.UrlOptions & RequestRetryOptions;
+	type Options = OptionsWithUri | OptionsWithUrl;
```
Comment (andy-ms) [inline types/requestretry/index.d.ts]: Where is this used?

File: types/requestretry/tslint.json (hunk 1 above)
Comment (EricByers) [inline types/requestretry/tslint.json]: Removed, was unneeded (reset both tslint and tsconfig by regenerating).

File: types/requestretry/index.d.ts (hunk 2 above)
Comment (EricByers) [inline types/requestretry/index.d.ts]: It is no longer used, removed all 3.

**Pull request merged**